### ISSUE TITLE
One-line change for Babashka support

### ISSUE THREAD

**Issue by matthewdowney** [HealthSamurai/matcho#7]
Hey there, thanks for this cool library. 

## Issue

I was hoping to get it running on Babashka, and I ran into trouble asserting literal values. E.g. asserting that `:x` is `:x`:

```clojure
$ bb -Sdeps '{:deps {healthsamurai/matcho {:mvn/version "0.3.9"}}}' \
     -e "(do (require 'matcho.core) (matcho.core/assert :x :x))"

----- Error --------------------------------------------------------------------
Type:     java.lang.Exception
Message:  Unable to resolve spec: :x
Location: resource:/src/babashka/clojure/spec/alpha.clj:91:11
```

(Whereas on JVM Clojure the result is of course `true`.) 

## Proposed solution

I think I found a one-line change to make the behavior consistent across the two platforms. 

This check is `true` on Babashka for just about everything and `false` on JVM Clojure, but I think the intent is kept if the check to `instance?` is replaced by just checking if the pattern is a spec, via `s/spec?`:

https://github.com/HealthSamurai/matcho/blob/20388b31e543cd6d9aed45a30519551f6bb66e8e/src/matcho/core.clj#L10-L12

I've confirmed that tests still pass after this change. Would you consider a PR for this?

**Comment by KGOH** [comment]
@matthewdowney applied in 8b8b852e1eb2c650ad1cd930d025014bf9855b80. Released 0.3.10, update is available on clojars. Please close the issue if the fix is working for you

**Comment by matthewdowney** [comment]
@KGOH That did the trick, thank you very much!
